python/cuspatial/cuspatial/geometry/geoarrowbuffers.py

```diff
@@ -0,0 +1,640 @@
+                    data.lines.z,
+                    data_locale=data_locale,
+                )
+            if data.polygons is not None:
+                self._polygons = PolygonArray(
+                    data.polygons.xy,
+                    data.polygons.polys,
+                    data.polygons.rings,
+                    data.polygons.mpolys,
+                    data.polygons.z,
+                    data_locale=data_locale,
+                )
+        else:
+            raise TypeError(
+                f"Invalid type passed to GeoArrowBuffers ctor {type(data)}"
+            )
+
+    @property
+    def points(self):
+        """
+        A simple numeric column. x and y coordinates are interleaved such that
+        even coordinates are x axis and odd coordinates are y axis.
+        """
+        return self._points
+
+    @property
+    def multipoints(self):
+        """
+        Similar to the Points column with the addition of an offsets column.
+        The offsets column stores the comparable sizes and coordinates of each
+        MultiPoint in the GeoArrowBuffers.
+        """
+        return self._multipoints
+
+    @property
+    def lines(self):
+        """
+        Contains the coordinates column, an offsets column, and a
+        mlines column. The mlines column  is optional. The mlines column stores
+        the indices of the offsets that indicate the beginning and end of each
+        MultiLineString segment. The absence of an `mlines` column indicates
+        there are no `MultiLineStrings` in the data source, only `LineString`s.
+        """
+        return self._lines
+
+    @property
+    def polygons(self):
+        """
+        Contains the coordinates column, a rings column specifying
+        the beginning and end of every polygon, a polygons column specifying
+        the beginning, or exterior, ring of each polygon and the end ring.
+        All rings after the first ring are interior rings.  Finally a
+        mpolygons column stores the offsets of the polygons that should be
+        grouped into MultiPolygons.
+        """
+        return self._polygons
+
+    def __len__(self):
+        """
+        The numer of unique geometries stored in this GeoArrowBuffers.
+        """
+        points_length = len(self._points) if self.points is not None else 0
+        lines_length = len(self._lines) if self.lines is not None else 0
+        multipoints_length = (
+            len(self._multipoints) if self.multipoints is not None else 0
+        )
+        polygons_length = (
+            len(self._polygons) if self.polygons is not None else 0
+        )
+        return (
+            points_length + lines_length + multipoints_length + polygons_length
+        )
+
+    def copy(self, deep=True):
+        """
+        Create a copy of all of the GPU-backed data structures in this
+        GeoArrowBuffers.
+        """
+        result = GeoArrowBuffers(self)
+        return result
```

Comment:
Looks like we're not handling the `deep` parameter here?

Reply:
Last minute oversight, thanks.